### PR TITLE
optional provide info about token owner

### PR DESCRIPTION
**Pull request by uniquename**
like discussed here #429

### CODE REVIEW

**Comment by e0ipso** [comment]
> just thought the same about api/me, but it would be another request. I think I will go for more information on api/login-token.

I think that making an extra request is fine in this case. We don't want to:
- Put in all the data we can in a very basic endpoint. Other use cases may not need all the user data back from the token endpoint.
- Complicate with even more configuration options.

I would just add the `uid` in the token endpoint.

File: modules/restful_token_auth/plugins/restful/restful_token_auth/token_auth/access_token/1.0/RestfulAccessTokenAuthentication.class.php [hunk 1]
```diff
@@ -59,7 +59,21 @@ public function getOrCreateToken() {
 
     $output = $this->viewEntity($id);
 
+    // various options to provide info about the token owner
+    $owner_resource = variable_get('restful_token_auth_owner_resource', '');
+
+    if ($owner_resource !== '<none>') {
+      if ($owner_resource == '<id>') {
+        $output['owner'] = $access_token->uid;
```
Comment (e0ipso) [inline modules/restful_token_auth/plugins/restful/restful_token_auth/token_auth/access_token/1.0/RestfulAccessTokenAuthentication.class.php:67]: Just keep this line.

**Comment by uniquename** [comment]
I'm using this module with a mobile app, thats why I'm like concerned about extra requests and I had thoughts like, why sending the owner id if there is no endpoint to get more information... that's why I came up with all that configuration... for me line 70 is the solution.

**Comment by e0ipso** [comment]
Thanks for getting back @uniquename. I am concerned about the module itself.

I understand, and share, your concern in your particular use case.

My recommendation would be to:
- Update this PR to make it simpler :smile: 
- Create `v2` (in a custom module) of the token endpoint. In that I would inherit from `RestfulAccessTokenAuthentication` and overwrite `getOrCreateToken` to inject your logic.
- Write a recipe of how you did that in the recipe book :-P (https://www.drupal.org/node/2380679)

I know I'm asking a lot, but I think that it's a good think to spread Open Source knowledge.

Thanks for your efforts!!

**Comment by uniquename** [comment]
Ok, I will see when I find time for that. Specially writing a recipe about it... cause that section really needs some love ;-)

File: modules/restful_token_auth/plugins/restful/restful_token_auth/token_auth/access_token/1.0/RestfulAccessTokenAuthentication.class.php (hunk 1 above)
Comment (amitaibu) [inline modules/restful_token_auth/plugins/restful/restful_token_auth/token_auth/access_token/1.0/RestfulAccessTokenAuthentication.class.php:67]: or even better, we can declare the owner in the `\RestfulTokenAuthenticationBase::publicFieldsInfo`, probably something like this:

``` php
    if (variable_get('restful_enable_users_resource', TRUE)) {
      $public_fields['owner'] = array(
        'property' => 'uid',
        'resource' => array(
          'user' => array(
            'name' => 'users',
            'full_view' => FALSE,
          ),
        ),
      );
    }
    else {
      $public_fields['owner'] = array(
        'property' => 'uid',
      );
    }
```

**Comment by uniquename** [comment]
Ok, kept on thinking about this for a while... and: just close the pull request ;-) I see just too many ifs and thens for different use cases... one other thought, shouldn't it be possible to enable/disable the csrf token endpoint? For mobile apps it's just not needed. And actually the same for the token endpoints, so one could enable the module to have the classes without having the endpoint.

**Comment by e0ipso** [comment]
@uniquename you can disable the endpoint if you set `hook_menu = FALSE` in the plugin definition and you don't provide a menu item. That will leave the endpoint available to use programatically but not hittable from the exterior world.

I believe that there must be a `hook_somethig_ctools_something_alter` to alter the plugin definition without having to hack `restful_token_auth`.

That would also be a great recipe :wink: wink.

**Comment by uniquename** [comment]
Then I would go for closing the endpoint with the htaccess ;-) ...sorry, joking... but why should it go this way for the token endpoint, when user and cookie endpoint are configurable in the backend? Like the comment in the your code says "By checking the variable, we allow implementing modules to easily implement their own "users" resource."

File: modules/restful_token_auth/plugins/restful/restful_token_auth/token_auth/access_token/1.0/RestfulAccessTokenAuthentication.class.php (hunk 1 above)
Comment (e0ipso) [inline modules/restful_token_auth/plugins/restful/restful_token_auth/token_auth/access_token/1.0/RestfulAccessTokenAuthentication.class.php:67]: @uniquename does this provide the info that you need?

**Comment by e0ipso** [comment]
I'm doing triage. @uniquename I will close this issue since it seems to be stalled.

Do not hesitate to re-open it.

Thanks for contributing!

**Pull request closed without merging**